source/fundamentals/auth.txt

```diff
@@ -0,0 +1,384 @@
+  default value ``admin``.
+* ``awsSessionToken`` - value your ``AWS_SESSION_TOKEN`` *(optional)*
+
+Select the :guilabel:`Connection String` or the :guilabel:`MongoCredential`
+tab below for instructions and sample code for specifying this authentication
+mechanism:
+
+.. tabs::
+
+   .. tab::
+      :tabid: Connection String
+
+      To specify the ``MONGODB-AWS`` authentication mechanism using a
+      connection string, pass the ``MONGODB-AWS`` value as a parameter
+      of ``authMechanism`` in your connection string. Your code to instantiate
+      a ``MongoClient`` should resemble the following:
+
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>&authMechanism=MONGODB-AWS");
+
+      If you need to specify an AWS session token, include it in the
+      ``authMechanismProperties`` parameter as follows:
+
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>&authMechanism=MONGODB-AWS&authMechanismProperties=AWS_SESSION_TOKEN:<awsSessionToken>");
+
+
+   .. tab::
+      :tabid: MongoCredential
+
+      To specify the default authentication mechanism using the
+      ``MongoCredential`` class, use the
+      :java-docs:`createAwsCredential() </apidocs/mongodb-driver-core/com/mongodb/MongoCredential.html#createAwsCredential(java.lang.String,char%5B%5D)>`
+      method. Your code to instantiate a ``MongoClient`` should resemble the
+      following:
+
+      .. include:: /includes/fundamentals/code-snippets/auth-credentials-aws.rst
+
+      If you need to specify an AWS session token, use the
+      :java-docs:`applyConnectionString() </apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyConnectionString(com.mongodb.ConnectionString)>`
+      method to specify a connection string that specifies the
+      ``authMechanism`` and ``AWS_SESSION_TOKEN`` parameter values.
+
+      .. include:: /includes/fundamentals/code-snippets/auth-credentials-aws-session.rst
+
+.. _x509-auth-mechanism:
+
+``X.509``
+~~~~~~~~~
+
+.. note::
+   The X.509 authentication mechanism is only available in MongoDB versions
+   2.6 and later.
+
+The ``X.509`` authentication mechanism uses
+:wikipedia:`TLS <Transport_Layer_Security>` with X.509 certificates to
+authenticate your user, identified by the distinguished name (DN) of your
+client certificate. For more information on determining the *subject* name from
+the X.509 certificate, see the MongoDB manual
+:manual:`X.509 tutorial </tutorial/configure-x509-client-authentication/#add-x-509-certificate-subject-as-a-user>`.
+
+The following code snippets show how to specify the connection mechanism,
+using the following placeholders:
+
+* ``username`` - your MongoDB username, derived from the certificate subject
```

Comment:
I think your previous mention of how to define the username:
```
For more information on determining the *subject* name from
the X.509 certificate, see the MongoDB manual
:manual:`X.509 tutorial </tutorial/configure-x509-client-authentication/#add-x-509-certificate-subject-as-a-user>`.
```
is somewhat easy to glance over by mistake. I think the link might be more discoverable if you linked the text "derived from the certificate subject" to that tutorial instead.

Reply:
Ok, would you mind if I linked it from both? I want to retain mention of what information they provide to use the specific mechanism which I did with SCRAM and AWS.